cider/exec/template/CaseIR.cpp

```diff
@@ -198,28 +203,34 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenCaseExpr(
     cgen_state_->ir_builder_.SetInsertPoint(then_bb);
     const auto then_expr_ptr = codegen(expr_pair.second.get(), co, true);
     auto then_expr = dynamic_cast<FixedSizeColValues*>(then_expr_ptr.get());
+    CHECK(then_expr);
     auto then_bb_lvs = then_expr->getValue();
+    auto then_bb_null_lvs = then_expr->getNull();
     if (is_real_str) {
       // FIXME(haiwei): [POAE7-2457] ,blocking by [POAE7-2415]
       CIDER_THROW(CiderCompileException,
                   "String type case when is not currently supported.");
     } else {
       then_lvs.push_back(then_bb_lvs);
+      then_null_lvs.push_back(then_bb_null_lvs);
```

Comment:
Can you add some comments to address how the null_lvs determined in CaseIR?

Reply:
Rewrite these code for NOT NULL columns.